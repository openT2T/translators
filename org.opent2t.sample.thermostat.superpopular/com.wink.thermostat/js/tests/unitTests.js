var test = require('ava');
var translatorPath = require('path').join(__dirname, '..');
var runThermostatTests = require('opent2t-device-thermostat/thermostatTests');
var deviceData = require('./devicedata');
var testData = require('./testdata');
var OpenT2TLogger = require('opent2t').Logger;
var logger = new OpenT2TLogger("info");
var MockHub = require('opent2t-device-winkhub/mockWinkHub');
var mockHub = new MockHub(logger, deviceData);

var settings = {
    logger,
    test,
    createTranslator: mockHub.createTranslator(translatorPath, deviceData.base_state.data.object_id, deviceData.base_state.data.uuid),
    setTestData: mockHub.setTestData,
    expectedExceptions : testData.expected_exceptions
};

// Run standard thermostat tests
runThermostatTests(settings);
