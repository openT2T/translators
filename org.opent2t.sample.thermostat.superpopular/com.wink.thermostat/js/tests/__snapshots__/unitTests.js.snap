exports[`GetPlatform 1`] = `"{\"opent2t\":{\"schema\":\"org.opent2t.sample.thermostat.superpopular\",\"translator\":\"opent2t-translator-com-wink-thermostat\",\"controlId\":\"D5D37EB6-F428-41FA-AC5D-918F084A4C93\",\"uuid\":\"D5D37EB6-F428-41FA-AC5D-918F084A4C93\"},\"availability\":\"online\",\"pi\":\"D5D37EB6-F428-41FA-AC5D-918F084A4C93\",\"mnmn\":\"Wink\",\"mnmo\":\"Thermostat (Generic)\",\"n\":\"Mock Name\",\"rt\":[\"org.opent2t.sample.thermostat.superpopular\"],\"entities\":[{\"n\":\"Mock Name\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"rt\":[\"opent2t.d.thermostat\"],\"di\":\"d8354f3e-ea15-f4ec-fe34-4b9a57c3397f\",\"resources\":[{\"href\":\"/ambientTemperature\",\"rt\":[\"oic.r.temperature\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]},{\"href\":\"/targetTemperature\",\"rt\":[\"oic.r.temperature\"],\"if\":[\"oic.if.a\",\"oic.if.baseline\"]},{\"href\":\"/targetTemperatureHigh\",\"rt\":[\"oic.r.temperature\"],\"if\":[\"oic.if.a\",\"oic.if.baseline\"]},{\"href\":\"/targetTemperatureLow\",\"rt\":[\"oic.r.temperature\"],\"if\":[\"oic.if.a\",\"oic.if.baseline\"]},{\"href\":\"/awayMode\",\"rt\":[\"oic.r.mode\"],\"if\":[\"oic.if.a\",\"oic.if.baseline\"]},{\"href\":\"/ecoMode\",\"rt\":[\"oic.r.sensor\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]},{\"href\":\"/hvacMode\",\"rt\":[\"oic.r.mode\"],\"if\":[\"oic.if.a\",\"oic.if.baseline\"]},{\"href\":\"/hasFan\",\"rt\":[\"oic.r.sensor\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]},{\"href\":\"/fanActive\",\"rt\":[\"oic.r.sensor\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]}]}]}"`;

exports[`GetPlatformExpanded 1`] = `"{"opent2t":{"schema":"org.opent2t.sample.thermostat.superpopular","translator":"opent2t-translator-com-wink-thermostat","controlId":"D5D37EB6-F428-41FA-AC5D-918F084A4C93","uuid":"D5D37EB6-F428-41FA-AC5D-918F084A4C93"},"availability":"online","pi":"D5D37EB6-F428-41FA-AC5D-918F084A4C93","mnmn":"Wink","mnmo":"Thermostat (Generic)","n":"Mock Name","rt":["org.opent2t.sample.thermostat.superpopular"],"entities":[{"n":"Mock Name","icv":"core.1.1.0","dmv":"res.1.1.0","rt":["opent2t.d.thermostat"],"di":"d8354f3e-ea15-f4ec-fe34-4b9a57c3397f","resources":[{"href":"/ambientTemperature","rt":["oic.r.temperature"],"if":["oic.if.s","oic.if.baseline"],"id":"ambientTemperature","temperature":50,"units":"c"},{"href":"/targetTemperature","rt":["oic.r.temperature"],"if":["oic.if.a","oic.if.baseline"],"id":"targetTemperature","temperature":50,"units":"c"},{"href":"/targetTemperatureHigh","rt":["oic.r.temperature"],"if":["oic.if.a","oic.if.baseline"],"id":"targetTemperatureHigh","temperature":100,"units":"c"},{"href":"/targetTemperatureLow","rt":["oic.r.temperature"],"if":["oic.if.a","oic.if.baseline"],"id":"targetTemperatureLow","temperature":0,"units":"c"},{"href":"/awayMode","rt":["oic.r.mode"],"if":["oic.if.a","oic.if.baseline"],"id":"awayMode","modes":["home"],"supportedModes":["home","away"]},{"href":"/ecoMode","rt":["oic.r.sensor"],"if":["oic.if.s","oic.if.baseline"],"id":"ecoMode","value":false},{"href":"/hvacMode","rt":["oic.r.mode"],"if":["oic.if.a","oic.if.baseline"],"id":"hvacMode","supportedModes":["off"],"modes":["off"]},{"href":"/hasFan","rt":["oic.r.sensor"],"if":["oic.if.s","oic.if.baseline"],"id":"hasFan","value":true},{"href":"/fanActive","rt":["oic.r.sensor"],"if":["oic.if.s","oic.if.baseline"],"id":"fanActive","value":false}]}]}"`;
